### PR TITLE
Refactor code to fix module_utils sanity, update changelog and initial unit test cases

### PR DESCRIPTION
**Pull request by ganeshrn**
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/704

##### SUMMARY

Refactor code to fix module_utils sanity, update changelog and initial unit test cases

##### ISSUE TYPE

- Docs Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -1,8 +1,10 @@
+ansible
```
Comment (pabelanger) [inline test-requirements.txt]: please drop this from requirements file. We don't actually need ansible, as we set it up by default.

Comment (ganeshrn) [reply test-requirements.txt]: Fixed

**Comment by pabelanger** [comment]
recheck

**Comment by ganeshrn** [comment]
recheck

**Comment by ganeshrn** [comment]
recheck

**Comment by ganeshrn** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by ganeshrn** [comment]
recheck

**Pull request merged**